registry/session_v2.go

```diff
@@ -19,10 +20,27 @@ func getV2Builder(e *Endpoint) *v2.URLBuilder {
 	return e.URLBuilder
 }
 
-func (r *Session) V2RegistryEndpoint(index *IndexInfo) (ep *Endpoint, err error) {
-	// TODO check if should use Mirror
-	if index.Official {
-		ep, err = newEndpoint(REGISTRYSERVER, true)
+func (r *Session) V2RegistryEndpoint(index *IndexInfo, useMirror bool) (ep *Endpoint, err error) {
+	if useMirror && len(index.Mirrors) > 1 {
+		for i := 0; i < len(index.Mirrors) && ep == nil; i++ {
+			ep, err = newEndpoint(index.Mirrors[0], index.Secure)
+			if err != nil {
+				return
+			}
+			err = validateEndpoint(ep)
+			if err != nil {
+				return
+			}
+			if ep.Version == APIVersion1 {
+				ep = nil
```

Comment:
Each mirror should be checked until a v2 registry is found.  If none of the mirrors are v2 enabled then the error is thrown.  We cannot know this until we have attempted to validate each of the endpoints.  Using a break on the else case of `if ep.Version == APIVersion1` is probably more legible than ` && ep == nil`, I will make that update.